lib/takeaway.rb

```diff
@@ -0,0 +1,43 @@
+require 'twilio-ruby'
+
+account_sid = 'sid'
+auth_token = 'token'
+@client = Twilio::REST::Client.new api_key_sid, api_key_secret, account_sid
+
+class Takeaway
+
+  attr_reader :MENU, :order
+  MENU = { "brocolli" => 3, "risotto" => 10, "bread" => 5 }
+
+  def initialize
+    @order = []
+    @menu = MENU
+    @order_total = 0
+  end
+
+  def read_menu
+    MENU
+  end
+
+  def select(pick)
+    @order << MENU.select { |key| key == pick }
+  end
+
+  def total
+    @order.each { |selection|
+      selection.each_value { |value| @order_total += value }
+    }
+    p @order_total
+  end
+
+  def check_total
+    total
+    "Your order was #{@order}. Your total was #{@order_total}."
+  end
+
+  @client.messages.create(
```

Comment:
Nice work with the Twilio gem! 👍

Reply:
But I suspect that you would like to know how to stop the tests from sending you texts ;)

Reply:
To achieve this, you'd need to use dependency injection and then, instead of injecting the real client, in your test you can use a double.

Reply:
The first part would look like this...

```ruby

class Takeaway

  attr_reader :MENU, :order
  MENU = { "brocolli" => 3, "risotto" => 10, "bread" => 5 }

  def initialize(twilio_client)
    @order = []
    @menu = MENU
    @order_total = 0
    @client = client
  end
```